rainyview/src/main/java/me/samlss/view/RainyView.java

```diff
@@ -439,112 +430,115 @@ public void release() {
+            rainDrop.speedY = rainDrop.speedX * Math.abs(rainDrop.slope);
 
-        rainDrop.mxLength = (float) Math.abs(Math.cos(degree * Math.PI / 180) * rainDropLength);
-        rainDrop.myLength = (float) Math.abs(Math.sin(degree * Math.PI / 180) * rainDropLength);
+            float rainDropLength = (float) mRainDropMinLength
+                    + mOnlyRandom.nextInt(mRainDropMaxLength - mRainDropMinLength);
+            double degree = Math.toDegrees(Math.atan(rainDrop.slope));
 
-        //random x coordinate
-        rainDrop.mx = mRainRect.left + mOnlyRandom.nextInt((int) mRainRect.getWidth());
-        //the fixed y coordinate
-        rainDrop.my = mRainRect.top - rainDrop.myLength;
+            rainDrop.mxLength = (float) Math.abs(Math.cos(degree * Math.PI / 180) * rainDropLength);
+            rainDrop.myLength = (float) Math.abs(Math.sin(degree * Math.PI / 180) * rainDropLength);
 
-        mRainDrops.add(rainDrop);
+            //random x coordinate
+            rainDrop.mx = mRainRect.left + mOnlyRandom.nextInt((int) mRainRect.getWidth());
+            //the fixed y coordinate
+            rainDrop.my = mRainRect.top - rainDrop.myLength;
 
-    }
+            mRainDrops.add(rainDrop);
 
-    /**
-     * Update all the raindrops state.
-     * */
-    private void updateRainDropState() {
-        mRemovedRainDrops.clear();
+        }
 
-        for (RainDrop rainDrop : mRainDrops) {
-            if (rainDrop.my - rainDrop.myLength > mRainRect.bottom) {
-                mRemovedRainDrops.add(rainDrop);
-                recycle(rainDrop);
-            } else {
-                if (rainDrop.slope >= 0) {
-                    rainDrop.mx += rainDrop.speedX;
+        /**
+         * Update all the raindrops state.
+         * */
+        private void updateRainDropState() {
+            mRemovedRainDrops.clear();
+
+            for (RainDrop rainDrop : mRainDrops) {
+                if (rainDrop.my - rainDrop.myLength > mRainRect.bottom) {
+                    mRemovedRainDrops.add(rainDrop);
+                    recycle(rainDrop);
                 } else {
-                    rainDrop.mx -= rainDrop.speedX;
+                    if (rainDrop.slope >= 0) {
+                        rainDrop.mx += rainDrop.speedX;
+                    } else {
+                        rainDrop.mx -= rainDrop.speedX;
+                    }
+                    rainDrop.my += rainDrop.speedY;
                 }
-                rainDrop.my += rainDrop.speedY;
             }
-        }
 
-        if (!mRemovedRainDrops.isEmpty()) {
-            mRainDrops.removeAll(mRemovedRainDrops);
+            if (!mRemovedRainDrops.isEmpty()) {
+                mRainDrops.removeAll(mRemovedRainDrops);
+            }
+
+            if (!mRainDrops.isEmpty()) {
+                invalidate();
+            }
         }
 
-        if (!mRainDrops.isEmpty()) {
-            invalidate();
+        @Override
+        public void run() {
+            createRainDrop();
```

Comment:
might throw IllegalArgumentException as per the method definition. Surround it with try/catch

Reply:
Done